py/dml/dmlc.py

```diff
@@ -153,6 +154,60 @@ def print_cdep(outputbase, headers, footers):
     f.close()
     f.commit()
 
+def dotdot_depth(path):
+    '''return how many levels of parent directories are required to resolve a
+    relative path'''
+    return os.path.normpath(path).count('..' + os.path.sep)
+
+assert dotdot_depth('./../../foo/../../../bar.dml') == 4
```

Comment:
This is weird. Is this really the right place for this test?

Reply:
I sometimes keep extremely lightweight tests of pure functions as top-level assertions to avoid boilerplate. In particular, I think this is a good way to make top-level `re.compile()` calls more readable. And this function has similar weight as a regexp.

Reply:
Hm, alright. I suppose it makes no difference, even if perhaps it's not "proper."